internal/search/query/transformer.go

```diff
@@ -163,6 +163,62 @@ func substituteOrForRegexp(nodes []Node) []Node {
 	return new
 }
 
+// substituteConcat reduces a concatenation of patterns to a separator-separated string.
+func substituteConcat(nodes []Node, separator string) []Node {
+	isPattern := func(node Node) bool {
+		if pattern, ok := node.(Pattern); ok && !pattern.Negated {
+			return true
+		}
+		return false
+	}
+	new := []Node{}
```

Comment:
nit: We want at least as many as previous number of nodes (if this is true).
```suggestion
	new := make([]Node, 0, len(nodes))
```